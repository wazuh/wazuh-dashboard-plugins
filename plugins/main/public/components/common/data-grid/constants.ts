import { PaginationOptions } from './types';

export const MAX_ENTRIES_PER_QUERY = 10000;
export const DEFAULT_PAGE_INDEX = 0;
export const DEFAULT_PAGE_SIZE = 15;
export const DEFAULT_PAGE_SIZE_OPTIONS = [DEFAULT_PAGE_SIZE, 25, 50, 100];
export const DEFAULT_PAGINATION_OPTIONS: PaginationOptions = {
  pageIndex: DEFAULT_PAGE_INDEX,
  pageSize: DEFAULT_PAGE_SIZE,
  pageSizeOptions: DEFAULT_PAGE_SIZE_OPTIONS,
};
