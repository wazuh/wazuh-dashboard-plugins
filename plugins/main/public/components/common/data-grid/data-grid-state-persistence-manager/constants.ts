export const KEY_STATE = {
  COLUMN: 'column',
  COLUMN_WIDTH: 'column-width',
  PAGE_SIZE: 'page-size',
} as const;

export type KeyState = (typeof KEY_STATE)[keyof typeof KEY_STATE];
