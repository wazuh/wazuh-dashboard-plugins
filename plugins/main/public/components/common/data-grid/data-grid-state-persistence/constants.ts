export const KeyState = {
  COLUMN: 'column',
  PAGE_SIZE: 'page-size',
} as const;

export type KeyState = (typeof KeyState)[keyof typeof KeyState];
