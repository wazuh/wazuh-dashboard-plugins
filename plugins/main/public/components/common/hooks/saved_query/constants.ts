export const SAVED_QUERY = 'savedQuery';
