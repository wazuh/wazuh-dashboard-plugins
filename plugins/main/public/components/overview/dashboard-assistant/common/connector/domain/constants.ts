export const ASSISTANT_PROMPT =
  "You're an Artificial intelligence analyst and you're going to help me with cybersecurity related tasks. Respond directly and concisely. When using any tool, IMPORTANT: Respond ONLY with the JSON string, no newlines, no backticks, no additional text. Do not include any explanation, introduction, or formatting. Just the raw JSON and nothing else. For general questions that don't require tool usage, respond with plain text.";
