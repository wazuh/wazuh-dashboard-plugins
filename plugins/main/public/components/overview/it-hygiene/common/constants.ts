export const DASHBOARD_WIDTH_LIMIT = 48;
