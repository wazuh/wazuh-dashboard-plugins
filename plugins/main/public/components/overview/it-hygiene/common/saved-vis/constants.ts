export const STYLE = {
  bgFill: '#000',
  bgColor: false,
  labelColor: false,
  subText: '',
  fontSize: 40,
} as const;

export const HEIGHT = 9;
