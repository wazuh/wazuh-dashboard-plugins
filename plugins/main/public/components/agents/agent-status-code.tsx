import React from 'react';
import { AGENT_STATUS_CODE } from '../../../common/constants';
import { ColumnWithStatusIcon } from './column-with-status-icon';

interface SyncedProps {
  statusCode: number;
}

interface StatusCodeAgent {
  STATUS_CODE: number;
  COLOR: string;
  STATUS_DESCRIPTION: string;
}

export const AgentStatusCode = ({ statusCode }: SyncedProps) => {
  const statusCodeAgent = AGENT_STATUS_CODE.find(
    (status: StatusCodeAgent) => status.STATUS_CODE === statusCode,
  );

  return (
    <ColumnWithStatusIcon
      color={statusCodeAgent?.COLOR ?? 'subdued'}
      text={statusCodeAgent?.STATUS_DESCRIPTION ?? 'Without information'}
    />
  );
};
