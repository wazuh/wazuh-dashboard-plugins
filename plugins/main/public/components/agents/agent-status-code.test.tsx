import React from 'react';
import { render } from 'enzyme';
import { AgentStatusCode } from './agent-status-code';
import { AGENT_STATUS_CODE } from '../../../common/constants';

interface StatusCodeAgent {
  STATUS_CODE: number;
  COLOR: string;
  STATUS_DESCRIPTION: string;
}

describe('AgentStatus component', () => {
  test.each(
    AGENT_STATUS_CODE.map(statusCode => ({
      STATUS_CODE: statusCode.STATUS_CODE,
      COLOR: statusCode.COLOR,
      STATUS_DESCRIPTION: statusCode.STATUS_DESCRIPTION,
    })),
  )(
    'Renders status code indicator with the its color and the text - %j',
    (input: StatusCodeAgent) => {
      const wrapper = render(
        <AgentStatusCode statusCode={input.STATUS_CODE} />,
      );

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('svg').prop('class')).toContain(
        `euiIcon--${input.COLOR}`,
      );
      expect(wrapper[1].children[0].data).toEqual(input.STATUS_DESCRIPTION);
    },
  );
});
