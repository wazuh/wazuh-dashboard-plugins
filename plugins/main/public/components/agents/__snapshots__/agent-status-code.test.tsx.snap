// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentStatus component Renders status code indicator with the its color and the text - {"STATUS_CODE":0,"COLOR":"success","STATUS_DESCRIPTION":"Agent is connected"} 1`] = `
Array [
  <span
    class="euiToolTipAnchor"
  >
    <div
      class="euiHealth euiHealth--textSizeS wz-flex"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon--success euiIcon-isLoading"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          />
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        />
      </div>
    </div>
  </span>,
  <span
    class="hide-agent-status"
  >
    Agent is connected
  </span>,
]
`;

exports[`AgentStatus component Renders status code indicator with the its color and the text - {"STATUS_CODE":1,"COLOR":"danger","STATUS_DESCRIPTION":"Invalid agent version"} 1`] = `
Array [
  <span
    class="euiToolTipAnchor"
  >
    <div
      class="euiHealth euiHealth--textSizeS wz-flex"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon--danger"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="8"
              cy="8"
              r="4"
            />
          </svg>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        />
      </div>
    </div>
  </span>,
  <span
    class="hide-agent-status"
  >
    Invalid agent version
  </span>,
]
`;

exports[`AgentStatus component Renders status code indicator with the its color and the text - {"STATUS_CODE":2,"COLOR":"danger","STATUS_DESCRIPTION":"Error retrieving version"} 1`] = `
Array [
  <span
    class="euiToolTipAnchor"
  >
    <div
      class="euiHealth euiHealth--textSizeS wz-flex"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon--danger"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="8"
              cy="8"
              r="4"
            />
          </svg>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        />
      </div>
    </div>
  </span>,
  <span
    class="hide-agent-status"
  >
    Error retrieving version
  </span>,
]
`;

exports[`AgentStatus component Renders status code indicator with the its color and the text - {"STATUS_CODE":3,"COLOR":"danger","STATUS_DESCRIPTION":"Shutdown message received"} 1`] = `
Array [
  <span
    class="euiToolTipAnchor"
  >
    <div
      class="euiHealth euiHealth--textSizeS wz-flex"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon--danger"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="8"
              cy="8"
              r="4"
            />
          </svg>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        />
      </div>
    </div>
  </span>,
  <span
    class="hide-agent-status"
  >
    Shutdown message received
  </span>,
]
`;

exports[`AgentStatus component Renders status code indicator with the its color and the text - {"STATUS_CODE":4,"COLOR":"danger","STATUS_DESCRIPTION":"Disconnected because no keepalive received"} 1`] = `
Array [
  <span
    class="euiToolTipAnchor"
  >
    <div
      class="euiHealth euiHealth--textSizeS wz-flex"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon--danger"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="8"
              cy="8"
              r="4"
            />
          </svg>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        />
      </div>
    </div>
  </span>,
  <span
    class="hide-agent-status"
  >
    Disconnected because no keepalive received
  </span>,
]
`;

exports[`AgentStatus component Renders status code indicator with the its color and the text - {"STATUS_CODE":5,"COLOR":"danger","STATUS_DESCRIPTION":"Connection reset by manager"} 1`] = `
Array [
  <span
    class="euiToolTipAnchor"
  >
    <div
      class="euiHealth euiHealth--textSizeS wz-flex"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon--danger"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="8"
              cy="8"
              r="4"
            />
          </svg>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        />
      </div>
    </div>
  </span>,
  <span
    class="hide-agent-status"
  >
    Connection reset by manager
  </span>,
]
`;
